src/utils/task_stats_alloc/tests/loom.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(loom)]
+
+/// Run `RUSTFLAGS="--cfg loom" cargo test --test loom` to test.
+use loom::sync::atomic::{AtomicBool, Ordering};
+use loom::sync::Arc;
+use loom::thread;
+use task_stats_alloc::TaskLocalBytesAllocated;
+
+#[test]
+fn test_to_avoid_double_drop() {
+    loom::model(|| {
+        let bytes_num = 3;
+        let num = Arc::new(TaskLocalBytesAllocated::new_for_test(3));
+
+        // Add the flag value when counter drop so we can observe.
+        let flag = Arc::new(AtomicBool::new(false));
+
+        let threads: Vec<_> = (0..bytes_num)
+            .map(|_| {
+                let num = num.clone();
+                let flag = flag.clone();
+                thread::spawn(move || {
+                    num.sub_for_test(1, flag);
```

Comment:
I would suggest to make it more complicated to include `add` method.

Reply:
It's hard. if we have thread call add, then it's possible that the counter reaches < 0.

Reply:
Why the counter reaches < 0? I think if all threads are the same pattern as `add; wait some time; sub` It still should only reaches 0 onces, otherwise it's broken.

Reply:
Ohh yes. Reaches < 0 if you spawn a thread only do add.

Your suggestions is good, I can have a try.